### PR TITLE
Make sure HOME, WORKDIR, and MPI env vars are set correctly.

### PR DESCRIPTION
**Pull request by luca-heltai**
These changes are required to avoid various warnings, and to match user and workdir for the `Dockerfile.root` image.

### CODE REVIEW

File: contrib/docker/Dockerfile.root [hunk 1]
```diff
@@ -1,5 +1,9 @@
...
 
-LABEL maintainer="luca.heltai@gmail.com"
-
 USER root
+
+ENV HOME /root
+
+WORKDIR /root
+
+ENV OMPI_ALLOW_RUN_AS_ROOT 1
```
Comment (jppelteret) [inline contrib/docker/Dockerfile.root:9]: After a quick search, I've spotted some references to `OMPI_ALLOW_RUN_AS_ROOT_CONFIRM=1`. Do we perhaps need this as well?

Comment (bangerth) [reply contrib/docker/Dockerfile.root:9]: Yes, I believe that the conclusion of the discussion we had with the OpenMPI folks was that you had to set both environment variables.

Comment (luca-heltai) [reply contrib/docker/Dockerfile.root:9]: Indeed. Added.

**Review by masterleinad — APPROVED** [review]

**Review by jppelteret — APPROVED** [review]

**Comment by jppelteret** [comment]
Let's not hold you up any more with this @luca-heltai. You're the expert here, not us :-D

**Pull request merged**